app.js

```diff
@@ -1,17 +1,36 @@
 const express = require("express");
 const { retrieveTopics, incorrectPathNames } = require("./controllers/topics.controllers");
 const { retrieveAllPathInfo } = require("./controllers/api.controllers")
+const { retrieveArticleById } = require("./controllers/articles.controllers")
 
 const app = express();
 
-// app.use(express.json());
+app.use(express.json());
 
 app.get("/api/topics", retrieveTopics);
 
 app.get("/api", retrieveAllPathInfo)
 
+app.get("/api/articles/:article_id", retrieveArticleById)
+
 app.all("*", incorrectPathNames);
 
+app.use((err, req, res, next) => {
+    if(err.status === 404){
+        res.status(404).send(({msg: err.msg}))
+    } else {
+        next (err)
+    }
+})
+
+app.use((err, req, res, next) => {
+    if(err.code === '22P02'){
+        //because error messages for PSQL ERRORS are not defined in the models, we do it here:
+        res.status(400).send(({msg: "Bad request"}))
+    }
```

Comment:
Very nice error handling, after this you could add in another error handler to handle `500` errors and console log them if they occur. `500` errors are internal and cover bugs in your code such as typeErrors or referenceErrors, they don't need to be sent to the client but are very useful for you to see as developer of the API